### PR TITLE
Feature: UCI API Onboarding

### PR DESCRIPTION
**Pull request by ChakshuGautam**

### CODE REVIEW

File: ansible/roles/kong-api/defaults/main.yml [hunk 1]
```diff
@@ -6552,3 +6554,617 @@ kong_apis:
...
+    uris: "{{ uci_admin_prefix }}/admin/v1/bot/get"
+    upstream_url: "{{ uci_admin_service_url }}/admin/v1/bot/get"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'solutionAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:7002]: Incorrect ACL as per API master sheet

File: ansible/roles/kong-api/defaults/main.yml [hunk 2]
```diff
@@ -6552,3 +6554,617 @@ kong_apis:
...
+    uris: "{{ uci_admin_prefix }}/admin/v1/bot/search"
+    upstream_url: "{{ uci_admin_service_url }}/admin/v1/bot/search"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'solutionAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:7056]: Incorrect ACL as per API master sheet

File: ansible/roles/kong-api/defaults/main.yml [hunk 3]
```diff
@@ -6552,3 +6554,617 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: registryUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:7154]: This is is missing some data in API master sheet

File: ansible/roles/kong-api/defaults/main.yml [hunk 4]
```diff
@@ -6552,3 +6554,617 @@ kong_apis:
       config.limit_by: credential
     - name: request-size-limiting
       config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+
+# UCI Related
+  - name: getUserSegmentsUCI
+    uris: "{{ uci_admin_prefix }}/admin/v1/userSegment/get"
+    upstream_url: "{{ uci_admin_service_url }}/admin/v1/userSegment/get"
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:6562]: Missing variable `uci_admin_service_url` in this definition file. This will result in ansible errors

File: ansible/roles/kong-api/defaults/main.yml [hunk 5]
```diff
@@ -6552,3 +6559,599 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: updateUserSegmentsUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:6601]: Missing in master sheet

File: ansible/roles/kong-api/defaults/main.yml [hunk 6]
```diff
@@ -6552,3 +6559,599 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: getAdapterUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:6691]: Missing in master sheet

File: ansible/roles/kong-api/defaults/main.yml [hunk 7]
```diff
@@ -6552,3 +6559,599 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: uploadFormsTransformerUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:6853]: Duplicate API

File: ansible/roles/kong-api/defaults/main.yml [hunk 8]
```diff
@@ -6552,3 +6559,599 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: getAllUserForBotUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:7105]: Missing in master sheet

File: ansible/roles/kong-api/defaults/main.yml [hunk 9]
```diff
@@ -6552,3 +6559,599 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: gqlLocationUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:7123]: Missing

File: ansible/roles/kong-api/defaults/main.yml [hunk 10]
```diff
@@ -6552,3 +6559,599 @@ kong_apis:
...
+      config.whitelist:
+        - 'userAdmin'
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ medium_rate_limit_per_hour }}"
+      config.limit_by: credential
+    - name: request-size-limiting
+      config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: gqlUCI
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:7141]: Missing

File: kubernetes/helm_charts/core/nginx-private-ingress/templates/configmap.yaml [hunk 11]
```diff
@@ -172,9 +172,24 @@ data:
...
+          rewrite ^/odk-service/(.*) /$1 break;
+          proxy_pass $target;
+        }
       }
     }
 kind: ConfigMap
 metadata:
   name: nginx-private-ingress.conf
-  namespace: {{ .Values.namespace }}
+  namespace: { { .Values.namespace } }
```
Comment (keshavprasadms) [inline kubernetes/helm_charts/core/nginx-private-ingress/templates/configmap.yaml:195]: Dont introduce unrelated changes please

**Review by keshavprasadms — CHANGES_REQUESTED** [review]
Please check comments

File: ansible/roles/kong-api/defaults/main.yml (hunk 6 above)
Comment (ChakshuGautam) [reply ansible/roles/kong-api/defaults/main.yml:6691]: Added

File: ansible/roles/kong-api/defaults/main.yml (hunk 7 above)
Comment (ChakshuGautam) [reply ansible/roles/kong-api/defaults/main.yml]: Removed

File: ansible/roles/kong-api/defaults/main.yml (hunk 8 above)
Comment (ChakshuGautam) [reply ansible/roles/kong-api/defaults/main.yml:7087]: Added.

File: ansible/roles/kong-api/defaults/main.yml (hunk 9 above)
Comment (ChakshuGautam) [reply ansible/roles/kong-api/defaults/main.yml]: Added

File: ansible/roles/kong-api/defaults/main.yml (hunk 10 above)
Comment (ChakshuGautam) [reply ansible/roles/kong-api/defaults/main.yml:7105]: Added

File: kubernetes/helm_charts/core/nginx-private-ingress/templates/configmap.yaml (hunk 11 above)
Comment (ChakshuGautam) [reply kubernetes/helm_charts/core/nginx-private-ingress/templates/configmap.yaml]: Apologies. Fixed.

File: ansible/roles/kong-api/defaults/main.yml (hunk 8 above)
Comment (keshavprasadms) [reply ansible/roles/kong-api/defaults/main.yml:7087]: I am unable to see this API in master sheet. Please help me with the row number

**Review by keshavprasadms — APPROVED** [review]
Looks good!
